scripts/gen-dockerfile.sh

```diff
@@ -1,66 +1,94 @@
-# TiKV root
-dir="."
-output="./Dockerfile"
+# This Docker image contains a minimal build environment for TiKV
+#
+# It contains all the tools necessary to reproduce official production builds of TiKV
+
+# We need to use CentOS 7 because many of our users choose this as their deploy machine.
+# Since the glibc it uses (2.17) is from 2012 (https://sourceware.org/glibc/wiki/Glibc%20Timeline)
+# it is our lowest common denominator in terms of distro support.
+
+# We require epel packages, so enable the fedora EPEL repo then install dependencies.
+cat <<EOT
+FROM centos:7.6.1810 as builder
```

Comment:
Why not `FROM pingcap/rust as builder`?

Reply:
There are a few reasons for this!

* The image was last updated 6 months ago, meaning it doesn't contain any updates since 6 months ago.
  + This is a security issue, as it means the build image hasn't received security updates!
 * Since we only state `pingcap/rust` there is no guarantee that the `pingcap/rust` image fetched will correspond with the `pingcap/tikv` version being built.
* The `pingcap/rust` image uses the wrong Rust version by default (and tries to prepackage it) creating a dependency on our tikv repo (since any time `rust-toolchain` changes we need to update that image too.
* The image uses a manually downloaded version of `tini` instead of relying on the one that Docker/k8s already provide.
* `pingcap/rust`'s only use seems to be `pingcap/tikv` meaning there is no benefit to abstracting it.
* Any user building the `pingcap/tikv` image repeatedly will already have the build steps in their cache, as these early steps won't need to be redone unless the `gen-dockerfile.sh` script gets changed.

As you can see, the steps required are not so complex (It's only a few lines) and it reduces the overall complexity of the system we must maintain.

In the future I *do* think we can improve the `pingcap/rust` image and start to use it again for a development build environment for TiKV.